scripts/CpUtil.lua

```diff
@@ -149,6 +125,35 @@ function CpUtil.getName(object)
 	return object.getName and object:getName() .. '/' .. helperName or 'Unknown'
 end
 
+function CpUtil.internalPrint(tag, ...)
+	CpUtil.try(
+		function (...)
+			local updateLoopIndex = g_updateLoopIndex and g_updateLoopIndex or 0
+			local timestamp = getDate( ":%S")
+			local infoString = string.format('%s [%s lp%d]', timestamp, tag or "---", updateLoopIndex)
+			print(string.format('%s %s', infoString, string.format(...)))
+		end, ...)
+end
+
+function CpUtil.internalPrintVehicle(vehicle, tag, str, ...)
+	str = string.format("%s: %s", CpUtil.getName(vehicle), str)
+	CpUtil.internalPrint(tag, str, ...)
+end
+
+function CpUtil.internalPrintImplement(implement, tag, str, ...)
+	str = string.format("%s: %s", CpUtil.getName(implement), str)
+	CpUtil.internalPrintVehicle(implement.rootVehicle, tag, str, ...)
+end
+
+-- convenience debug function that expects string.format() arguments,
+-- CpUtil.debugVehicle( CpDebug.DBG_TURN, "fill level is %.1f, mode = %d", fillLevel, mode )
+---@param channel number
+function CpUtil.debugFormat(channel, ...)
+	if CpDebug and CpDebug:isChannelActive(channel) then
+		CpUtil.internalPrint("dbg"..channel, ...)
```

Comment:
Since we are already at it, don't we want to get rid of this channel as a number, which isn't explicitly defined anywhere, I have no idea which number is which? Or replace it with short, meaningful string?